From 9e6c4445ef4e55bb23262b0ac2447b86320ebfbd Mon Sep 17 00:00:00 2001
From: Daniel Kurtz <djkurtz@chromium.org>
Date: Wed, 29 Jan 2014 00:29:12 +0800
Subject: [PATCH 1/2] glx_oml_sync_control: report FAIL if glXGetMscRateOML
 returns an error

The glx_oml_sync_control does not parse parameters.  In particular, it
does not parse the "-auto" flag to set piglit_automatic.

So, when we return FAIL from draw(), it returns to the
piglit_glx_event_loop, which just continues to spin waiting for events.
Instead, just report the FAIL result, which quits the program.

Signed-off-by: Daniel Kurtz <djkurtz@chromium.org>
---
 tests/spec/glx_oml_sync_control/getmscrate.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tests/spec/glx_oml_sync_control/getmscrate.c b/tests/spec/glx_oml_sync_control/getmscrate.c
index 9f0a43b..c18ea49 100644
--- a/tests/spec/glx_oml_sync_control/getmscrate.c
+++ b/tests/spec/glx_oml_sync_control/getmscrate.c
@@ -41,7 +41,7 @@ draw(Display *dpy)
 
 	if (!glXGetMscRateOML(dpy, win, &numerator, &denominator)) {
 		printf("glXGetMscRateOML returned failure.\n");
-		return PIGLIT_FAIL;
+		piglit_report_result(PIGLIT_FAIL);
 	}
 
 	if (numerator == 0xDEADBEEF) {
-- 
1.9.0.rc1.175.g0b1dcb5

